code/modules/mob/living/carbon/human/death.dm

```diff
@@ -12,11 +12,11 @@
 	if(buckled)
 		buckled.unbuckle_mob(src, force = TRUE)
 
-	Stun(100, TRUE, TRUE)//hold them still as they get deleted so they don't fuck up the animation
 	notransform = TRUE
+
+	QDEL_IN(src, 2 SECONDS) // since this is sometimes called in
 	dust_animation()
-	spawn_dust(just_ash)
-	QDEL_IN(src, 20) // since this is sometimes called in
+	addtimer(CALLBACK(src, PROC_REF(spawn_dust), just_ash), 1.5 SECONDS)//spawn the dust after the body disappears
```

Comment:
This is adding a timer to something that is expected to be qdel'ed, not a good idea.
You should keep how it was earlier, I don't think it was the problem here.